### DISCUSSION TITLE
2.8.10 更新日志

### DISCUSSION THREAD

**Discussion by LTCatt** [Meloong-Git/PCL#5071] [讨论]
## 整合包导入优化
- 支持安装同时包含 modpack 文件和启动器的懒人包
    在导入这种压缩文件包时，会自动换用被打包的启动器进行后续安装。
    因此，玩家即使用自己的 PCL 导入整合包，也能看到设置的背景图片、主页和音乐。
    下图为目前最推荐的懒人包分发格式：
    ![image](https://github.com/user-attachments/assets/dc66d20b-586e-4237-9c48-0797720a73ee)
- 支持识别、导入嵌套了多层文件夹的压缩文件（#4950，@NorthZeroD）
- 功能隐藏选项中添加了禁止 Mod 更新的设置，以避免整合包玩家误更新 Mod（#5033，@E0x72-21）
- 解压、复制文件时，可以准确地显示操作进度了
- 开始安装整合包时，会自动跳转到下载管理页面
- 导入压缩文件包成功后，会自动跳转到版本选择页面
- 导入压缩文件包或自动导入整合包时，会自动设置默认名称，不再弹窗要求输入
- 自动导入整合包时，会将导入后的文件夹添加到文件夹列表，且不再确认是否导入

## 其他更新内容
- Mod 详情页面会按 Mod 加载器分类卡片（#2456，@xiao-qwq）
![image](https://github.com/user-attachments/assets/595a6a2e-9db4-44f7-abca-af347338c8f9)
- 优化 Mixin 相关崩溃的分析（#4321，@lingyunawa）
- 优化了没有有效的服务端时的提示信息（#4982，@z0z0r4）
- 支持分析 Mod 不兼容的错误报告（#4980，@MCSteve123）
- 更新资源下载页面的版本列表筛选（#4996，@Louis-Harsune，PR @3gf8jv4dv）
- 为 Minecraft Experience 披风添加翻译（#5029，@tiantian114514，PR @3gf8jv4dv）
- 收藏的版本不再在其他列表中重复显示（#5031，@Crotina）
- 按下 F5 可以刷新任何可刷新的页面（#5042，PR @ZeroWolf233）
- 禁止从回收站拖拽安装 Mod 文件，以防止文件名错误
- 由于已无法访问，移除 OptiFine 中文镜像站的链接与下载源
- 不再尝试对 OptiFine 官方源和 MCIM 镜像源进行多线程下载
- 由于 Mod 数据库存在缺失，安装整合包时不再尝试从 MCIM 镜像源获取 Mod 列表

## 技术性调整
- 按 F5 可以刷新本地帮助页面；虽不会重新下载联网帮助页面，但也会重载它（#4423，@Joker2184）
- `MyImage.EnableCache` 换用 `DependencyProperty` 实现，以规避潜在的访问性问题

## Bug 修复
- 偶发的多线程集合操作冲突（#3771，@lactobionicAcid）
- 下载页左边栏无法完全显示时缺失滚动条（#4224，@Louis-Harsune）
- 无法从镜像源下载 OptiFine 1.8 和 1.9（#4281，@Silverteal）
- 崩溃报告中的日志有可能乱码（#4700，@wuxiangzhicao）
- 从隐藏主题切换回普通主题时，顶部栏背景图片没有清空（#4960，@lactobionicAcid）
- 对 MCIM 镜像源进行了多线程下载（#4981，@tangge233）
- 在特定情况下 Mod 管理页面部分功能异常（#4992，@Ender-Elaina-Theater）
- 若在修改版本隔离时的警告弹窗中选择取消，版本隔离实际上依然被修改了（#4995，@Vector-Syobon-812，PR @MiyamizuSu）
- 使用 Gitee 的主页预设无法加载（#5011，@MFn233）
- 自动搜索 Java 不会搜索名称仅包含数字的文件夹（#5016，@freecube0209）
- 除非重启启动器，否则正版登录不会自动刷新（#5053，@FQO3）
- 部分 Mod 的缓存信息可能有误，导致图标无法加载（#5062，@Hoicey522）
- 使用 MCIM 源安装 CurseForge 整合包时可能提示缺失 Mod
- `MyImage` 检查缓存过期时的行为有误，导致它不一定会使用缓存
- 部分 Mixin 失败的崩溃分析给出了错误的 Mod 结果
- 导入压缩文件包结束后没有任何提示

## 下个版本的计划
- 完成除 #4758 以外的所有中优先度及更高的项
- 做一部分低优先度 issue

<hr /><em>This discussion was created from the release <a href='https://github.com/Hex-Dragon/PCL2/releases/tag/2.8.10'>2.8.10</a>.</em>

**Comment by Joker2184** [comment]
更新日志主页已更新...

**Comment by kong-que-dnf123** [comment]
细节：
对于更新速报方面增强了一点“激动”性
![image](https://github.com/user-attachments/assets/eb4d48ba-821b-4a8f-aee7-c4a421279eb3)

**Reply by ZeroWolf233** [reply]
> 细节： 对于更新速报方面增强了一点“激动”性 ![image](https://private-user-images.githubusercontent.com/120994373/384394812-eb4d48ba-821b-4a8f-aee7-c4a421279eb3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwNzQ3MzIsIm5iZiI6MTczMTA3NDQzMiwicGF0aCI6Ii8xMjA5OTQzNzMvMzg0Mzk0ODEyLWViNGQ0OGJhLTgyMWItNGE4Zi1hZWU3LWM0YTQyMTI3OWViMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwOFQxNDAwMzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMTljMjg5NmVhZmI4YjY3MDBhMmUwYjVkMmE5Y2JjNTU2OGFjZDI1MjIwZmE2OWI2MDE4YWFkZDNkYzMxZjk4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.kkUsaWa59KAJYQwK--FZyt-bvP4KjyyX9IRmkwcHNMY)

# 更新！冲！

**Reply by SqrtTan** [reply]
冲刺，冲刺，冲！(otto音

**Comment by lanOST1** [comment]
mod安装方面可以加入GitHub和gitee上发布的mod，也可以下载

**Reply by tangge233** [reply]
emmm
除非有现成维护的列表，否则没有那么多精力去做这一个功能…

**Reply by LinQingYuu** [reply]
>mod安装方面可以加入GitHub和gitee上发布的mod，也可以下载

显然我们并不知道哪些是 Mod 哪些不是
并且 GitHub 连通性并不优秀
有些 Mod 并没有提供 Release
相较比花费时间和精力去维护没有啥大用处的列表，我觉得还是写两个新功能比较好

**Reply by WForst-Breeze** [reply]
已有类似反馈，#2288 已拒绝
